x-pack/test/licensing_plugin/apis/changes.ts

```diff
@@ -0,0 +1,134 @@
+  const PageObjects = getPageObjects(['common', 'security']);
+
+  const scenario = {
+    async setup() {
+      await security.role.create('license_manager-role', {
+        elasticsearch: {
+          cluster: ['all'],
+        },
+        kibana: [
+          {
+            base: ['all'],
+            spaces: ['*'],
+          },
+        ],
+      });
+
+      await security.user.create('license_manager_user', {
+        password: 'license_manager_user-password',
+        roles: ['license_manager-role'],
+        full_name: 'license_manager user',
+      });
+
+      // ensure we're logged out so we can login as the appropriate users
+      await PageObjects.security.logout();
+      await PageObjects.security.login('license_manager_user', 'license_manager_user-password');
+    },
+
+    async teardown() {
+      await security.role.delete('license_manager-role');
+    },
+
+    async startBasic() {
+      const response = await esSupertestWithoutAuth
+        .post('/_license/start_basic?acknowledge=true')
+        .auth('license_manager_user', 'license_manager_user-password')
+        .expect(200);
+
+      expect(response.body.basic_was_started).to.be(true);
+    },
+
+    async startTrial() {
+      const response = await esSupertestWithoutAuth
+        .post('/_license/start_trial?acknowledge=true')
+        .auth('license_manager_user', 'license_manager_user-password')
+        .expect(200);
+
+      expect(response.body.trial_was_started).to.be(true);
+    },
+
+    async deleteLicense() {
+      const response = await esSupertestWithoutAuth
+        .delete('/_license')
+        .auth('license_manager_user', 'license_manager_user-password')
+        .expect(200);
+
+      expect(response.body.acknowledged).to.be(true);
+    },
+
+    async getLicense(): Promise<PublicLicenseJSON> {
+      // > --xpack.licensing.pollingFrequency set in test config
+      // to wait for Kibana server to re-fetch the license from Elasticsearch
+      await delay(1000);
+
+      const { body } = await supertest.get('/api/licensing/info').expect(200);
+      return body;
+    },
+  };
+
+  describe('changes in license types', () => {
+    after(async () => {
+      await scenario.startBasic();
+    });
+
+    it('provides changes in license types', async () => {
+      await scenario.setup();
+      const initialLicense = await scenario.getLicense();
+      expect(initialLicense.license?.type).to.be('basic');
+      // security enabled explicitly in test config
+      expect(initialLicense.features?.security).to.eql({
+        isAvailable: true,
+        isEnabled: true,
+      });
+
+      const refetchedLicense = await scenario.getLicense();
+      expect(refetchedLicense.license?.type).to.be('basic');
+      expect(refetchedLicense.signature).to.be(initialLicense.signature);
+
+      // server allows to request trial only once.
+      // other attempts will throw 403
+      await scenario.startTrial();
```

Comment:
Due to this, I keep everything in one test suit. Because I cannot switch the license to the trial again and it creates a "hidden" state, which can affect new tests in the future.